src/components/TodoAllChecked.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useContext, useState, useEffect } from 'react';
+import { TodoContext } from '../TodoContext';
+
+export const TodoAllChecked: React.FC = () => {
+  const { todos, setTodos } = useContext(TodoContext);
+  const [
+    isChecked,
+    setIsChecked,
+  ] = useState(todos.every(todo => todo.completed));
+
+  useEffect(() => {
+    if (todos.every(todo => todo.completed)) {
+      setIsChecked(true);
+    } else {
+      setIsChecked(false);
+    }
+  }, [todos]);
+
+  const handleToggleAllChange = () => {
+    if (isChecked) {
+      setIsChecked(false);
+    } else {
+      setIsChecked(true);
+    }
+
+    setTodos(todos.map(todo => (
+      { ...todo, completed: !isChecked }
+    )));
+  };
+
+  if (todos.length > 0) {
```

Comment:
```suggestion
  if (!!todos.length) {
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**